config.go

```diff
@@ -23,26 +23,67 @@ import (
 )
 
 const (
-	RecordCount = "recordCount"
-	ReadTime    = "readTime"
-	Fields      = "fields"
-	Format      = "format"
+	RecordCount   = "recordCount"
+	ReadTime      = "readTime"
+	FormatType    = "format.type"
+	FormatOptions = "format.options"
 
 	FormatRaw        = "raw"
 	FormatStructured = "structured"
+	FormatFile       = "file"
 )
 
-var knownFieldTypes = []string{"int", "string", "time", "bool"}
+var (
+	knownFieldTypes = []string{"int", "string", "time", "bool"}
+	requiredFields  = []string{FormatType, FormatOptions}
+)
+
+type RecordConfig struct {
+	FormatType    string
+	FormatOptions map[string]interface{}
```

Comment:
Yup, I think we may add a few more in future.

Reply:
I don't see how we would since the value comes from a single config field 😕 even when we parse name:type tokens (arguably multiple format options) we store them under a single key. But I may be missing something and I don't think it's a big issue anyway, let's just leave it as it is.

Reply:
The reason why I think there might be multiple options is because I'm not quite sure that using a single option will always be enough to fully configure a payload. So using a map now will be less changes later. But since we (or at least I) have no plans to do that very soon, it probably is better to just make it an interface{}.